src/core/core.tooltip.js

```diff
@@ -321,7 +330,9 @@ function getTooltipSize(tooltip, model) {
 	helpers.each(model.beforeBody.concat(model.afterBody), maxLineWidth);
 
 	// Body lines may include some extra width due to the color box
-	widthPadding = model.displayColors ? (bodyFontSize + 2) : 0;
+	if (tooltip._options.mode !== 'dataset') {
```

Comment:
I don't understand why you needed to change setting the `widthPadding`

Reply:
First, I applied the default widthPadding in tooltip._options.mode is 'dataset',
but, the boxes and labels string  have overlaped each other.